### ISSUE TITLE
Please add multicast support for OPA and verbs providers

### ISSUE THREAD

**Issue by medawsonjr** [ofiwg/libfabric#3301]
I'm a professional in the Financial Trading Industry, and am keenly interested in the progress of libfabric for OPA and verbs. However, as our industry makes heavy use of UDP multicast (e.g., exchange market data feeds), lack of multicast support renders libfabric virtually useless for us.

Opening this issue at the request of Sean Hefty:

https://github.com/ofiwg/libfabric/issues/2627

**Comment by joerns** [comment]
I would like to express my interest in this as well, in case the libfabric developers are looking for input what features are needed by users. In our case it's mostly the verbs provider and in the future hopefully psm2 that we are interested in. Thanks.

**Comment by fleapapa** [comment]
Seems like currently only `udp` provider supports multicast. Am i correct?

**Comment by shefty** [comment]
I believe so.  Tagging @a-ilango so that he's aware multicast over verbs is wanted.

**Comment by aingerson** [comment]
@mlefebvre1 The FI_COLLECTIVE API was added to libfabric and rxm implements it, which allows use with tcp and verbs. Would you be able to use it for your use case?
https://ofiwg.github.io/libfabric/main/man/fi_collective.3.html

**Comment by mlefebvre1** [comment]
I might be mistaken, but my understanding of the collective API is that it will handle scatter operations by doing point to point transfers rather than off-loading the scatter to the switch fabric like you would if you use IP multicast or UD send/recv multicast with Verbs.

**Comment by aingerson** [comment]
Yes, correct. It is not using IP multicast but rather doing point to point transfers but functionally it is similar to multicast where you can do a one to all broadcast. I was just wondering if that satisfies your functional needs in regards to the API since all the collective groups act as a multicast address. But yes, they aren't the exact same.

**Comment by j-xiong** [comment]
Close as "will not fix"
